scipy/interpolate/_cubic.py

```diff
@@ -330,3 +335,163 @@ def from_spline(cls, tck, extrapolate=None):
+        arbitrary number of dimensions, but the length along `axis` (see below)
+        must match the length of `x`.
+    axis : int, optional
+        Axis along which `y` is assumed to be varying. Meaning that for
+        ``x[i]`` the corresponding values are ``np.take(y, i, axis=axis)``.
+        Default is 0.
+    extrapolate : bool, optional
+        Whether to extrapolate to ouf-of-bounds points based on first and last
+        intervals, or to return NaNs. Default is True.
+
+    Attributes
+    ----------
+    x : ndarray, shape (n,)
+        Breakpoints. The same `x` which was passed to the constructor.
+    c : ndarray, shape (4, n-1, ...)
+        Coefficients of the polynomials on each segment. The trailing
+        dimensions match the dimensions of `y`, excluding `axis`. For example,
+        if `y` is 1-d, then ``c[k, i]`` is a coefficient for
+        ``(x-x[i])**(3-k)`` on the segment between ``x[i]`` and ``x[i+1]``.
+    axis : int
+        Interpolation axis. The same `axis` which was passed to the
+        constructor.
+
+    Methods
+    -------
+    __call__
+    derivative
+    antiderivative
+    integrate
+    roots
+
+    See Also
+    --------
+    Akima1DInterpolator
+    PchipInterpolator
+
+    Notes
+    -----
+    .. versionadded:: 0.18.0
+
+    There are two additional equations required to determine all coefficients
+    of polynomials on each segment. They are formed from the conditions, that
+    at the first and the last interior knots the third derivative is
+    continuous. It essentially means that the first and the second segments
+    (on both ends) are described by the same cubic polynomial. These conditions
+    are called "not-a-knot" and are known to give better interpolation accuracy
+    when nothing is known about end point derivatives [1]_.
+
+    References
+    ----------
+    .. [1] de Boor, C., "A Practical Guide to Splines, Springer-Verlag", 1978.
+    """
+    def __init__(self, x, y, axis=0, extrapolate=True):
+        x, y = map(np.asarray, (x, y))
+        axis = axis % y.ndim
+        if x.ndim != 1:
+            raise ValueError("`x` must be 1-dimensional.")
+        if x.shape[0] < 2:
+            raise ValueError("`x` must contain at least 2 elements.")
+        if x.shape[0] != y.shape[axis]:
+            raise ValueError("The length of `y` along `axis`={0} doesn't "
+                             "match the length of `x`".format(axis))
+        if not np.all(np.isfinite(x)):
+            raise ValueError("`x` must contain only finite values.")
+        if not np.all(np.isfinite(y)):
+            raise ValueError("`y` must contain only finite values.")
+
+        dx = np.diff(x)
+        if np.any(dx <= 0):
+            raise ValueError("`x` must be strictly increasing sequence.")
+
+        # Find derivative values at each x[i] by solving a tridiagonal system.
+        # Cases n=2 and n=3 are solved by linear and quadratic interpolation.
```

Comment:
I meant that "not-a-knot" condition is not enough to determine derivatives in this case. I decided that the most natural way would be to assume that the solution is quadratic (or linear) function passing through the given points.